src/main/java/com/greenfoxacademy/goddesstribesbackend/controllers/BuildingController.java

```diff
@@ -0,0 +1,114 @@
+package com.greenfoxacademy.goddesstribesbackend.controllers;
+
+import com.greenfoxacademy.goddesstribesbackend.models.MockData;
+import com.greenfoxacademy.goddesstribesbackend.models.dtos.*;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class BuildingController {
+
+  @ApiImplicitParams({
+      @ApiImplicitParam(name = "token", value = "Authorization token",
+          required = true, dataType = "string", paramType = "header") })
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message ="OK", response = BuildingDTO.class)})
+  @GetMapping(value = "/kingdom/buildings",
+  produces = { "application/json" },
+  consumes = { "application/json" })
+  public ResponseEntity<Object> mockListOfBuildings() {
+    return ResponseEntity.status(200).body(MockData.buildingsDTO);
+  }
+
+  @ApiImplicitParams({
+      @ApiImplicitParam(name = "token", value = "Authorization token",
+          required = true, dataType = "string", paramType = "header") })
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message ="OK", response = BuildingDTO.class),
+      @ApiResponse(code = 400, message ="Missing parameter(s): type!", response = ErrorMessage.class),
+      @ApiResponse(code = 406, message ="Invalid building type", response = ErrorMessage.class),
+      @ApiResponse(code = 409, message ="Not enough resource", response = ErrorMessage.class)})
+  @PostMapping(value = "/kingdom/buildings",
+  produces = { "application/json" },
+  consumes = { "application/json" })
+  public ResponseEntity<Object> mockCreateABuilding (@RequestBody BuildingTypeDTO buildingTypeDTO) {
+    if (buildingTypeDTO.getType() == null || buildingTypeDTO.getType().isEmpty()) {
+      return ResponseEntity.status(400).body(new ErrorMessage("Missing parameter(s): type!"));
+    }
+    if (!(BuldingTypeENUM.FARM.toString().equalsIgnoreCase(buildingTypeDTO.getType())) &&
+        !(BuldingTypeENUM.MINE.toString().equalsIgnoreCase(buildingTypeDTO.getType())) &&
+        !(BuldingTypeENUM.BARRACK.toString().equalsIgnoreCase(buildingTypeDTO.getType())) )
+        {
+      return ResponseEntity.status(406).body(new ErrorMessage("Invalid building type"));
+    }
+    if (MockData.gold.getAmount() < 250) {
```

Comment:
Can we have these exported as constants, with nice naming in the front of the class please?

Reply:
Done